sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1050,19 +1032,38 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
   }
 
   /**
-   * Create a [[CatalogStorageFormat]] for creating tables.
+   * Parse the `createFileFormat` and put the storage information into the passed in map.
    *
-   * Format: STORED AS ...
+   * Example format:
+   * {{{
+   *   STORED AS INPUTFORMAT formatName1 OUTPUTFORMAT formatName2
+   * }}}
+   *
+   * OR
+   *
+   * {{{
+   *   STORED AS fileFormatName
+   * }}}
    */
-  override def visitCreateFileFormat(
-      ctx: CreateFileFormatContext): CatalogStorageFormat = withOrigin(ctx) {
+  private def getFileFormat(
+      ctx: CreateFileFormatContext,
+      storageProps: mutable.HashMap[String, String]): Unit = {
     (ctx.fileFormat, ctx.storageHandler) match {
       // Expected format: INPUTFORMAT input_format OUTPUTFORMAT output_format
       case (c: TableFileFormatContext, null) =>
-        visitTableFileFormat(c)
+        Option(string(c.inFmt)).foreach(inFmt => storageProps += "inputFormat" -> inFmt)
```

Comment:
`storageProps += "inputFormat" -> string(c.inFmt)`? `string(c.inFmt)` throws an NPE if it is null; so option is redundant...